test-alerts.cpp

```diff
@@ -1,8 +1,58 @@
 #define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
 
 #include "test/catch.hpp"
-#include "typewise-alert.h"
+#include "typewise-alert.hpp"
 
 TEST_CASE("infers the breach according to limits") {
   REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
+  REQUIRE(inferBreach(40, 20, 30) == TOO_HIGH);
+  REQUIRE(inferBreach(25, 20, 30) == NORMAL);
+}
+
+TEST_CASE("Test for classify Temperature breach") {
+	REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, MID_VALUE) == NORMAL);
+	REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, LOW_VALUE) == TOO_LOW);
+	REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, HIGH_VALUE) == TOO_HIGH);
+
+	REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, MID_VALUE) == NORMAL);
+	REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, LOW_VALUE) == TOO_LOW);
+	REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, HIGH_VALUE) == TOO_HIGH);
+
+	REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, MID_VALUE) == NORMAL);
+	REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, LOW_VALUE) == TOO_LOW);
+	REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, HIGH_VALUE) == TOO_HIGH);
+
+}
+
+TEST_CASE("Test for Conroller") {
+	BatteryCharacter batteryChar {PASSIVE_COOLING,"AMARON"};
+	checkAndAlert(TO_CONTROLLER, batteryChar, LOW_VALUE);
```

Comment:
These tests only fail when the program crashes. Real functionality can only be checked with real controller (hardware). Try testing the content of the communication to the controller.